tasks/pkg-suse.yml

```diff
@@ -24,31 +24,30 @@
     state: present
   when: not ansible_check_mode
 
+- set_fact:
+   zypper_repo_url: "{% if datadog_agent5 %}{{datadog_agent5_zypper_repo}}{% else %}{{datadog_zypper_repo}}{% endif %}"
+
+# ansible don't allow repo_gpgcheck to be set, we have to create the repo file manually
 - name: Install DataDog zypper repo
-  zypper_repository:
-    name: datadog
-    description: Datadog, Inc.
-    repo: "{{ datadog_zypper_repo }}"
-    enabled: yes
-    auto_import_keys: yes
-    disable_gpg_check: "{% if ansible_distribution_version|int == 11 %}yes{% else %}no{% endif %}" # SLES11 considers datadog's repo as untrusted
-    state: "{% if datadog_agent5 %}absent{% else %}present{% endif %}"
+  template:
+    src: zypper.repo.j2
+    dest: /etc/zypp/repos.d/datadog.repo
+    owner: "root"
+    group: "root"
+  register: result
 
-- name: (agent5) Install DataDog yum repo
-  zypper_repository:
-    name: datadog_5
-    description: Datadog, Inc.
-    repo: "{{ datadog_agent5_zypper_repo }}"
-    enabled: yes
-    auto_import_keys: yes
-    disable_gpg_check: "{% if ansible_distribution_version|int == 11 %}yes{% else %}no{% endif %}" # SLES11 considers datadog's repo as untrusted
-    state: "{% if datadog_agent5 %}present{% else %}absent{% endif %}"
+# refresh zypper repos only if the template changed
+- name: refresh zypper_repos
+  command: zypper refresh
```

Comment:
can we make the command refresh only the `datadog` repository?

Reply:
yes !